pkgs/tools/compression/lz4/default.nix

```diff
@@ -33,17 +34,27 @@ stdenv.mkDerivation rec {
     # TODO do this instead
     #"BUILD_STATIC=${if enableStatic then "yes" else "no"}"
     #"BUILD_SHARED=${if enableShared then "yes" else "no"}"
+    #"WINDRES:=${stdenv.cc.bintools.targetPrefix}windres"
   ]
     # TODO delete and do above
     ++ stdenv.lib.optional (enableStatic) "BUILD_STATIC=yes"
     ++ stdenv.lib.optional (!enableShared) "BUILD_SHARED=no"
+    ++ stdenv.lib.optional stdenv.hostPlatform.isMinGW "WINDRES:=${stdenv.cc.bintools.targetPrefix}windres"
+    # TODO make full dictionary
+    ++ stdenv.lib.optional stdenv.hostPlatform.isMinGW "TARGET_OS=MINGW"
     ;
 
   doCheck = false; # tests take a very long time
   checkTarget = "test";
 
-  # TODO remove
-  postInstall = stdenv.lib.optionalString (!enableStatic) "rm $out/lib/*.a";
+  # TODO(@Ericson2314): Make resusable setup hook for this issue on Windows.
```

Comment:
It looks like lz4 intentionally puts it in bindir. Perhaps the setup hook should instead add $out/bin to the `-L` lookups:

https://github.com/lz4/lz4/blob/49c20c498b80e70bae3ae604ef468921d79d2121/lib/Makefile#L182-L187

Reply:
This is something slightly awkward about windows, shared lirbaries (should) end up in `bin`. There are a few cases in nixpkgs as well I believe, where we specifically place shared libraries into `bin` for windows builds. That way (because the `bin` folders would end up on PATH, it usually just works on windows.

I keep traversing all dependencies, and collect `.dll`s from `lib` and `bin` and place them alongside the final executable, as that seems to be the most reliable way to build some form of distributable for windows.

Reply:
So the way I think of it is that the libraries primarily should live in `lib`, so that you don't pick up extra exe dependencies in the wrong multiple outputs, but each exe needs it's dlls symlinked in its `bin` directory since there is no `RUNPATH` or `RPATH` by which it could find them.

Reply:
For `mingw`, i think we should first import their pre-compiled binary packages.
It is trivial: as there is no `rpath` on Windows, there is no need to `patchelf` pre-compiled binaries. Just unpack and add few symlinks.
And then, the directory structure of `mingw` binaries will be "canonical" (as far as I remember there is no `lib/`) and the result of compilation from the sources has to be replacement for binary distribution.

Reply:
Err, why use pre-compiled anything? We are entirely building from source here (except for linux native boostrapping tools).

Reply:
I think my thing with the TODO is fine for now.